### ISSUE TITLE
Addon has registrating only first change condition of ipfs

### ISSUE THREAD

**Issue by AlexBeLi** [ipfs/ipfs-companion#259]
Firefox 55 (2017.05.31) new profile; IPFS Companion 2.0.0alpha6

Step to reproduction
	1. Start Ipfs
	2. Start Firefox
	3. Stop Ipfs
	4. Start Ipfs
![ipfs companion](https://cloud.githubusercontent.com/assets/20776735/26692200/3d5b1fcc-4719-11e7-8655-c9841445971c.png)

**Comment by lidel** [comment]
I was able to replicate this in the latest Nightly, but with these steps:

1. stop go-ipfs
2. start Firefox (icon is grey)
3. start IPFS (icon is still grey)
4. start/stop IPFS - icon istays grey, no API stats

It looks like a regression in Firefox Nightly. Fetching peer info hangs forever when API is down:

```javascript
async function getSwarmPeerCount () {
  try {
    const peerInfos = await ipfs.swarm.peers()
    return peerInfos.length
  } catch (error) {
    return -1
  }
```

(`await ipfs.swarm.peers()` hangs forever, while it should throw an error when API is offline)

I'll revisit this in a few days, perhaps it will get fixed upstream.

**Comment by lidel** [comment]
I retested this issue with current Nightly (`firefox-56.0a1`)  and was not able to replicate in any of the above scenarios.

I am assuming it was fixed upstream and closing this issue.

@AlexBeLi  If you are able to, please retest with latest nightly and confirm the issue is gone. 
If it is still happening, please reopen with more details (your OS, how you run `go-ipfs`, etc)

**Comment by AlexBeLi** [comment]
20170630030203
Mozilla/5.0 (Windows NT 6.3; Win64; x64; rv:56.0) Gecko/20100101 Firefox/56.0
Conditions of reproduce has changed.
1. Create new profile and install IPFS Companion 2.0.0alpha6
2. Start ipfs
3. _**Click on extension button and "Open WebUI"**_
4. Stop ipfs
State of icon not changing. If I click on extension button after that in browser console showing
"TypeError: can't access dead object"

If I don't do action from third point all works good, state of icon changed on inactive. 
Maybe it another bug?

**Comment by lidel** [comment]
Thank you. I was able to replicate it with steps you provided.

The `ipfs-api-status-update` alarm just stops being executed, which in turn does not trigger peer count update, so it is the same bug.

Reopening, will look into it in spare time.

**Comment by lidel** [comment]
I've just released [v2.0.0alpha7](https://github.com/ipfs/ipfs-companion/releases/tag/v2.0.0alpha7) which should fix this. 
Let me know if the issue is gone.

**Comment by AlexBeLi** [comment]
All works fine.
